cabby/model/landmark_recognition/dataset.py

```diff
@@ -0,0 +1,179 @@
+    region: The region of the data.
+    s2level: The s2level of the cells.
+  Returns:
+    The train, validate and test sets and the dictionary of labels to cellids.
+  '''
+  rvs_dataset = datasets.RVSDataset(data_dir, s2level, region)
+
+  train_dataset = TextGeoSplit(
+    rvs_dataset.train)
+  logging.info(
+    f"Finished to create the train-set with {len(train_dataset)} samples")
+  val_dataset = TextGeoSplit(
+    rvs_dataset.valid)
+  logging.info(
+    f"Finished to create the valid-set with {len(val_dataset)} samples")
+  test_dataset = TextGeoSplit(
+    rvs_dataset.test)
+  logging.info(
+    f"Finished to create the test-set with {len(test_dataset)} samples")
+
+  return train_dataset, val_dataset, test_dataset
+
+tag_values = ['O', "L"]
+tag_values.append("PAD")
+tag2idx = {t: i for i, t in enumerate(tag_values)}
+
+class TextGeoSplit(torch.utils.data.Dataset):
+  """A split of of the RUN dataset.
+
+  `points`: The ground true end-points of the samples.
+  `labels`: The ground true label of the cellid.
+  `cellids`: The ground truth S2Cell id.
+  `neighbor_cells`: One neighbor cell id of the ground truth S2Cell id.
+  `far_cells`: One far away cell id (in the region defined) of the ground truth
+  'dprob': Gamma distribution probability.
+  S2Cell id.
+  """
+
+  def __init__(self, data: pd.DataFrame):
+    # Tokenize instructions.
+
+    self.inputs = [
+      tokenize_and_preserve_labels(sent, labs)
+      for sent, labs in zip(data.instructions.tolist(), data.entity_span)
+    ]
+    self.sent = data.instructions.tolist()
+
+
+  def __getitem__(self, idx: int):
+    '''Supports indexing such that TextGeoDataset[i] can be used to get
+    i-th sample.
+    Arguments:
+      idx: The index for which a sample from the dataset will be returned.
+    Returns:
+      A single sample including text, the correct cellid, a neighbor cellid,
+      a far cellid, a point of the cellid and the label of the cellid.
+    '''
+
+    input = {k: torch.tensor(v) for k, v in self.inputs[idx].items()}
+    input['instructions'] = self.sent[idx]
+    return input
+
+  def __len__(self):
+    return len(self.inputs)
+
+
+def tokenize_and_preserve_labels(sentence, text_labels):
+  labels = []
+  spans = sorted(list(text_labels.values()))
+
+  sentence_words = word_tokenize(sentence)
+  cur_index = 0
+  span = spans.pop(0)
+  start, end = span[0], span[1]
+  for word in sentence_words:
+    if cur_index >= start:
+      labels.append(1)
+    else:
+      labels.append(0)
+
+    cur_index += len(word)
+    if cur_index<len(sentence) and sentence[cur_index] == ' ':
+      cur_index+=1
+    if cur_index >= end:
+      if len(spans)>0:
+        span = spans.pop(0)
+        start, end = span[0], span[1]
+      else:
+        start = len(sentence)+1
+
+  tokenized_input = tokenize_and_align_labels(labels, sentence_words)
+  return tokenized_input
+
+def tokenize_and_align_labels(tags, sentence_word, label_all_tokens = True):

```

Comment:
Documentation please! Example inputs and outputs much appreciated.

Reply:
There is documentation in the dataset_bert.py. I will remove dataset.py